src/or/hs_descriptor.c

```diff
@@ -1415,6 +1445,70 @@ encrypted_data_length_is_valid(size_t len)
   return 0;
 }
 
+/* Decrypt the descriptor cookie given the descriptor, the auth client,
+ * and the client secret key. On sucess, return 0 and a newly allocated
+ * descriptor cookie descriptor_cookie_out. On error or if the client id
+ * is invalid, return -1 and descriptor_cookie_out is set to
+ * NULL. */
+static int
+decrypt_descriptor_cookie(const hs_descriptor_t *desc,
+                          const hs_desc_authorized_client_t *client,
+                          const curve25519_secret_key_t *client_sk,
```

Comment:
`client_sk` is a fine variable name, but I wonder if we could make it more explicit that this is for optional client auth.
Something like `client_sk -> client_auth_sk`? What do you think?

Reply:
Agreed